qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1816,6 +1816,20 @@ def _safe_serialize_figures(self):
                     figures[name] = figure
         return figures
 
+    def _safe_serialize_data(self):
+        """Return serializable object for the experiment data"""
+        data_list = ThreadSafeList()
+        with self._data.lock:
+            for circuit_run_data_dict in self.data():
+                data = {}
+                for data_key, data_value in circuit_run_data_dict.items():
+                    if data_key == "counts":
```

Comment:
I think this is going to have pretty poor performance for large data if it has to iterate over every count dictionary. I tried running some JSON tests and it looks like `Count` objects in terra are serialized the same as a regular dict object. Could you check if removing this method and just letting the data get serialized as default works? And if not can you post what the warning/exceptions are?

Here was my test code:
```python
from qiskit.result import Counts

a = {"00": 10, "01": 20, "10": 30, "11": 40}
b = Counts(a)
c = dict(b)

# Test default JSON encoder
import json

for counts in [a, b, c]:
    obj = {"counts": counts}
    print(repr(json.dumps(obj)))

# Test qiskit-experiments JSON encoder
from qiskit_experiments.framework import ExperimentEncoder

for counts in [a, b, c]:
    obj = {"counts": counts}
    print(repr(json.dumps(obj, cls=ExperimentEncoder)))
```

Reply:
In `test_qubit_spectroscopy.py` in method `test_expdata_serialization`  I have a new test for checking that `experiment_data` is seriable:
```python
self.assertRoundTripSerializable(expdata, self.experiment_data_equiv)
```
If I am not using the new method, this test fails. I checked the object and it seems like the decoded object is missing data under 'counts':
![image](https://user-images.githubusercontent.com/51112651/168810176-b25f0fad-2766-4c49-8648-8f82d842cc51.png)

I think it worked for you because the dictionary you created is `Dict[str,int]` while in the experiment the dictionary is `Dict[str, numpy.int64]` and `numpy.int64` isn't serializable.

Reply:
Ah I see. In that case I think we need to add a case to the `ExperimentEncoder` to handle raw numpy numbers. Something like this should hopefully work:

```python
if isinstance(obj, numpy.number):
    return obj.item()
```

Reply:
Working great! changed as suggested.